pkg/config/mock/go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/DataDog/datadog-agent/pkg/config/mock
 
-go 1.22.0
+go 1.21.0
```

Comment:
We probably want to revert this change

Reply:
This is needed since OTEL still used `1.21` and we now use the mock in package used by OTEL. This is related to the `inv check-otel-module-versions` job